tests/src/JIT/Directed/VectorABI/VectorMgdMgdArray.cs

```diff
@@ -0,0 +1,1099 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.Intrinsics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// Test passing and returning HVAs (homogeneous vector aggregates) to/from managed code.
+// Test various sizes (including ones that exceed the limit for being treated as HVAs),
+// as well as passing various numbers of them (so that we exceed the available registers),
+// and mixing the HVA parameters with non-HVA parameters.
+
+// This Test case covers all cases for
+//   Methods that take one HVA argument with between 1 and 9 Vectror64 or Vector128 elements
+//   - Called normally or by using reflection
+//   Methods that return an HVA with between 1 and 9 Vectror64 or Vector128 elements
+//   - Called normally or by using reflection
+
+// Remaining Test cases to do:
+// - Add tests that have one than one HVA argument
+// - Add types that are *not* HVA types (e.g. too many vectors, or using Vector256).
+// - Add tests that use a mix of HVA and non-HVA arguments
+
```

Comment:
This comment should probably be updated to reflect that his test is focused on Arrays (and the "Remaining ..." comment is probably no longer relevant).
And it might be good to make it clear that these tests are testing that passing and returning HVAs to/from arrays, rather than passing and returning arrays of HVAs. I confess that I found it easy to confuse myself between the two.